tests/tests_firewall_fact.yml

```diff
@@ -0,0 +1,86 @@
+- name: Test firewall ansible fact
+  hosts: all
+  become: true
+  tasks:
+    - name: Test firewall ansible fact
+      block:
+        - name: Start with default configuration
+          include_role:
+            name: linux-system-roles.firewall
+          vars:
+            firewall:
+              - previous: replaced
+
+        - name: Get default ansible fact
+          include_role:
+            name: linux-system-roles.firewall
+          vars:
+            firewall:
+
+        - name: Fail if firewall_config is undefined
+          fail:
+            msg: Ansible fact firewall_config is not defined
+          when: firewall_config is undefined
+
+        - name: Fail if firewall_config custom has attributes
+          vars:
+            custom_item: "{{ firewall_config.custom[item] | list }}"
+          fail:
+            msg: "{{ custom_item }}"
+          when: custom_item | length != 0
+          loop: "{{ firewall_config.custom.keys() | list }}"
+
+        - name: Fail if firewall_config defaults do not have attributes
+          vars:
+            default_item: "{{ firewall_config.default[item] | list }}"
+          fail:
+            msg: default {{ item }} does not have attributes
+          when: default_item | length == 0 and item != "ipsets"
+          loop: "{{ firewall_config.default.keys() | list }}"
+
+        - name: save default ansible fact value
+          set_fact:
+            __previous_firewall_config: "{{ firewall_config }}"
+          no_log: true
+
+        - name: modify firewalld configuration
+          shell:
+            cmd: |
+              firewall-cmd --permanent --add-service https
+              firewall-cmd --permanent --new-service custom
```

Comment:
I think one of these is supposed to be a zone?  And does `https` exist by default?

Reply:
those are just to show that modified defaults present themselves in custom dictionary. The first one is to add https to the default zone's permanent configuration